modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptPropertyDefValGenerator.cs

```diff
@@ -163,19 +164,81 @@ INamedTypeSymbol symbol
                     continue;
                 }
 
-                // TODO: Detect default value from simple property getters (currently we only detect from initializers)
-
-                EqualsValueClauseSyntax? initializer = property.DeclaringSyntaxReferences
-                    .Select(r => r.GetSyntax() as PropertyDeclarationSyntax)
-                    .Select(s => s?.Initializer ?? null)
-                    .FirstOrDefault();
+                var propertyDeclarationSyntax = property.DeclaringSyntaxReferences
+                    .Select(r => r.GetSyntax() as PropertyDeclarationSyntax).FirstOrDefault();
 
                 // Fully qualify the value to avoid issues with namespaces.
                 string? value = null;
-                if (initializer != null)
+                if (propertyDeclarationSyntax?.Initializer != null)
```

Comment:
I would prefer both this `if` block and the `if (string.IsNullOrWhiteSpace(value))` block below to be inside a `if (propertyDeclarationSyntax != null)` block. Seems clearer to me when reading, and the generated code may be better. I haven't checked this, but I think the current one would do unnecessary additional checks when null (`string.IsNullOrWhiteSpace(value)` and `propertyGet != null` below).

Reply:
With the Initializer, it is indeed impossible to have the latter, and here the code should be changed.